Make style changes to 1533 (#2109)
temporalio/documentation: docs-src/concepts/what-is-worker-versioning.md

We recommend that you read about Workflow Definitions before proceeding, because Workflow Versioning is largely concerned with helping to manage nondeterministic changes to those definitions.

Worker Versioning helps manage nondeterministic changes by providing a convenient way to ensure that [Workers](#workers) with different Workflow and Activity Definitions operating on the same [Task Queue](#task-queue) don't attempt to process [Workflow Tasks](#workflow-task) and [Activity Tasks](#activity-task-execution) that they can't successfully process, according to sets of versions associated with that Task Queue that you've defined.

Accomplish this goal by assigning a Build ID (a free-form string) to the code that defines a Worker, and specifying which Build IDs are compatible with each other by updating the version sets associated with the Task Queue, stored by the Temporal Server.

### When and why should you use Worker Versioning

> The main reason to use this feature is to deploy incompatible changes to short-lived [Workflows](/workflows). When you use this feature on Task Queues, the Workflow starter doesn't have to know about the introduction of new versions.

Reviewer: This one needs to stay "On Task Queues using this feature" I think. Both phrasings are accurate in some sense, but I think the important thing here is that Task Queues which use the feature act with different behavior

If you want to change an Activity Definition's type signature while creating a new incompatible Build ID for a Worker, you can do so without worrying about the Activity failing to execute on some other Worker with an incompatible definition. The same principle applies to Child Workflows.

:::tip

Public facing Workflows on a versioned Task Queue shouldn't change their signature as it contradicts the purpose of Workflow-launching Clients remaining unaware of changes in the Workflow Definition. If you need to change a Workflow's signature, use a different Workflow Type or a completely new Task Queue.

:::

:::note

If you schedule an Activity or a Child Workflow on _a different_ Task Queue from the one the Workflow runs on, the system doesn't assign a specific version. This means if the target queue is versioned, they run on the latest default, and if its unversioned, they operate as they would have without this feature.

:::

**Continue-As-New and Worker Versioning**

By default, a versioned Task Queue's Continue-as-New function starts the continued Workflow on the same compatible set as the original Workflow.

> If you continue-as-new onto a different Task Queue, the system doesn't assign any particular version. You also have the option to specify that the continued Workflow should start using the Task Queue's latest default version.

Reviewer: The behavior here changed a bit, but I'll reword it in my main PR after we merge this

### How to use Worker Versioning

> To use this Worker Versioning:

Reviewer: ```suggestion
To use Worker Versioning:
```

1. Define Worker build-identifier version sets for the Task Queue.
   You can use either the `temporal` CLI or your choice of SDK.
2. Enable the feature on your Worker by specifying a Build ID.

#### Defining the version sets

Whether you use `temporal` [CLI](/cli/) or an SDK, updating the version sets feels the same. You
specify the Task Queue that you're targeting, the Build ID that you're adding (or
promoting), whether it becomes the new default version, and any existing versions it should be
considered compatible with.

The rest of this section uses updates to one Task Queue's version sets as examples.

> By default, both Task Queues and Workers are in an unversioned state. [Unversioned Worker](#unversioned-workers can poll

Reviewer: ```suggestion
By default, both Task Queues and Workers are in an unversioned state. [Unversioned Worker](#unversioned-workers) can poll
```

config property.

### Build ID reachability

Eventually, you'll want to know if you can retire the old Worker versions. Temporal provides functionality
to help you determine if a version is still in by open or closed Workflows. You can use the
`temporal` CLI to do this with the following command:

```command
temporal task-queue get-build-id-reachability
```

The command tells if the Build ID in question is unreachable, only reachable by closed Workflows, or reachable by open and new Workflows on a per-Task-Queue basis.
For example, this "2.0" Build ID is shown here by the CLI to be reachable by both new Workflows and
some existing Workflows:

```command
temporal task-queue get-build-id-reachability --build-id "2.0"
```

```output
BuildId                         TaskQueue                                   Reachability
    2.0  build-id-versioning-dc0068f6-0426-428f-b0b2-703a7e409a97  [NewWorkflows
                                                                   ExistingWorkflows]
```

For more information, see the [CLI documentation](/cli/) or help output.

You can also use this API `GetWorkerTaskReachability` directly from within language SDKs.

> ### Unversioned Workers
> 
> Unversioned Workers refer to Workers operating on a Task Queue before versioning capabilities were added to that Task Queue.
> They don't have any specific version or Build IDs associated with them, hence they're in the _compatible set zero_, or unversioned state.

Reviewer: ```suggestion
### Unversioned Workers

Unversioned Workers refer to Workers that have not opted into the Worker Versioning feature in their configuration. They will only receive tasks from Task Queues which do not have any version sets defined on them, or which have open workflows that began executing before versions were added to the queue.
```
